Assets/Scripts/InteractionSystem/InteractableObjects/DoorBehavior.cs

```diff
@@ -0,0 +1,85 @@
+/******************************************************************
+*    Author: Marissa Moser
+*    Contributors: 
+*    Date Created: May 22, 2024
+*    Description: Contains the functionality for the door when it is interacted with.
+
+*******************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using PlaceboEntertainment.UI;
+
+public class DoorBehavior : MonoBehaviour, IInteractable
+{
+    private bool _isOpened = false;
+    [SerializeField] private bool _isLocked;
+    private Animator _anim;
+
+    void Start()
+    {
+        _anim = GetComponent<Animator>();
+    }
+
+    /// <summary>
+    /// Called from player's Interact script. Contains functionality for when
+    /// a door is interacted with. Can be updated to also unlock the door.
+    /// </summary>
+    /// <param name="player"></param>
+    public void Interact(GameObject player)
+    {
+        if (!_isLocked && !_isOpened)
+        {
+            OpenDoor();
+        }
+    }
+
+    /// <summary>
+    /// Displays the specific UI prompt if the door is locked or unlocked
+    /// </summary>
+    public void DisplayInteractUI()
+    {
```

Comment:
If the door is open already, this function doesn't need to execute correct? I'd add a guard clause, would make this slightly easier to read.